### DISCUSSION TITLE
conjure_time_step_wizard! error

### DISCUSSION THREAD

**Discussion by Swarnali-D** [CliMA/Oceananigans.jl#3597] [Computational science!]
I was trying out the Langmuir Turbulence example, which gave me an error while setting the cfl criteria as:
`julia> conjure_time_step_wizard!(simulation, cfl=1.0, max_Δt=1minute)`

`ERROR: UndefVarError: 'conjure_time_step_wizard!' not defined`

There is an irregular use of this command across the examples. Every example uses `TimeStepWizard` while only Langmuir Turbulence, Convecting Plankton and Baroclinic Adjustment uses `conjure_time_step_wizard!`

How do I overcome the error?

**Comment by glwagner** [comment]
What version of `Oceananigans` are you using

**Reply by Swarnali-D** [reply]
I am using Oceananigans v0.90.6

**Reply by glwagner** [reply]
Ok, why don't you try to upgrade to the latest version and see if you still have the error?

**Reply by Swarnali-D** [reply]
Yes, Upgrading worked. Thanks!

**Reply by glwagner** [reply]
Awesome. The reason you were getting the error is because, once upon a time, `conjure_time_step_wizard!` did not exist. But you must have been running the example from `main` or something. The lesson is that code and software are not compatible, in general, unless you deliberately make them compatible.

**Reply by Swarnali-D** [reply]
So what features do `conjure_time_step_wizard!` have that `TimeStepWizard` did not?

**Reply by glwagner** [reply]
It doesn't have new features. It just builds a wizard and then adds it as a callback to simulation, reducing the number of lines required to use the `TimeStepWizard` from 2 to 1.

**Reply by Swarnali-D** [reply]
Ok thanks a lot!
